ArtOfIllusion/src/artofillusion/CameraFilterDialog.java

```diff
@@ -44,9 +45,12 @@ public class CameraFilterDialog extends BDialog implements RenderListener
   private static final int PREVIEW_WIDTH = 200;
   private static final int PREVIEW_HEIGHT = 150;
 
+  public boolean okPressed;
```

Comment:
What is this boolean doing? I see that you explicitly set it on lines 53 and 142, but it's not *read* anywhere.

Reply:
The way it _should_ have been done it would probably be to have the exit of the dialog return a boolean, but then it would have changed the the interface... And the entire process is so weird with the interactions between the filters, track and the camera parameters that it would take a much more thorough redesign of the process to make it look any clearer. So better to keep it in minimum this time.

If you'll look at the file, not just the changes it should explain thing a bit.

Reply:
I am not particularly happy with the structure either, but I'd rather keep this one just as a functional bug fix. It was a patch-over-patch but now it least now it works.

Reworking it will be a much bigger project and if we go into that it may never be finished --> users will never ge the working version. I also found a bunch of other things too that are wrong with the filters, but they don't prevet using the filter tracks. I'd hold those to the next round.